service/controller/legacy/v29/ebs/ebs.go

```diff
@@ -128,6 +128,61 @@ func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment Volu
 		if err != nil {
 			return microerror.Mask(err)
 		}
+	}
+
+	{
+		o := func() error {
+			i := &ec2.DescribeVolumesInput{
+				VolumeIds: []*string{
+					aws.String(volumeID),
+				},
+			}
+
+			v, err := e.client.DescribeVolumes(i)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if len(v.Volumes) == 0 {
+				// Fall through.
+				e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("volume %#q not found", volumeID))
+				return nil
+			}
+
+			if len(v.Volumes) > 1 {
+				// Fall through.
+				e.logger.LogCtx(ctx, "level", "debug", "message", "multiple volumes found")
+				return nil
+			}
+
+			d := v.Volumes[0]
+			if len(d.Attachments) > 1 {
+				// Shouldn't happen; log so we know if it is
+				e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found multiple attachment for volume %#q", volumeID))
+			}
+
+			for _, a := range d.Attachments {
+				if a.State != nil {
+					attachmentStatus := *a.State
+					if attachmentStatus != "detached" {
+						return microerror.New(fmt.Sprintf("the volume %#q is still attached", volumeID))
+					}
+				} else {
+					// Shouldn't happen; log so we know if it is
+					e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("nil attachment state for volume %#q", volumeID))
+				}
+			}
+
+			return nil
+		}
+
+		b := backoff.NewExponential(30*time.Second, 5*time.Second)
+		n := backoff.NewNotifier(e.logger, context.Background())
+
+		err := backoff.RetryNotify(o, b, n)
```

Comment:
That is the wrong approach because this blocks the resource and thus all other reconciliation, which is what we want to prevent. All the checking on the volumes can be done, but we should not backoff anymore within resources. When the detaching is not yet done, the resource should be canceled. It is then executed again anyway on the next reconciliation loop after the regular resync period.

Reply:
So what you recommend here it is simply to remove the backoff func and leave the logic like it is. Did I understand correctly?

Reply:
As discussed we want to remove the backoff magic here.